### PR TITLE
Upgrade Gunicorn to v23.0.0 and switch to gthread worker class

### PR DESCRIPTION
**Pull request by amitsrivastava**
## What changes were proposed in this pull request?

- This update includes switching the Gunicorn worker class from `eventlet` to `gthread`. The `gthread` worker class is included with Gunicorn and is sufficient for our current requirements.

- Previously, we used `eventlet` for their superior cooperative multitasking capabilities, which were crucial when running a single-process setup with 32–64 threads. However, with our current configuration of 32 independent worker processes, each running 2 threads, the complexity of `eventlet` and `gevent` is no longer justified. Concurrency is now limited to the two threads within each process, where `gthread` provides a simpler and adequate solution.

- This change simplifies our setup by reducing dependencies on external libraries and making future upgrades easier.

## How was this patch tested?

- (Please explain how this patch was tested. Ex: unit tests, manual tests)
- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Review by Harshg999 — COMMENTED** [review]
Nice! Left a review comment.

Also did you test this gunicorn upgrade E2E with a live setup?

File: desktop/core/base_requirements.txt [hunk 1]
```diff
@@ -80,7 +79,5 @@ drf-spectacular[sidecar]==0.27.2
 -e file://${ROOT}/desktop/core/ext-py3/boto-2.49.0
 -e file://${ROOT}/desktop/core/ext-py3/django-axes-5.13.0
 -e file://${ROOT}/desktop/core/ext-py3/djangosaml2-0.18.0
--e file://${ROOT}/desktop/core/ext-py3/dnspython-1.16.0  # Required for eventlet==0.30.2 and python3.10
--e file://${ROOT}/desktop/core/ext-py3/eventlet-0.30.2
```
Comment (Harshg999) [inline desktop/core/base_requirements.txt:84]: Can you also remove these packages from `/ext-py3` directory?

Comment (amitsrivastava) [reply desktop/core/base_requirements.txt:84]: done.

**Comment by amitsrivastava** [comment]
> Nice! Left a review comment.
> 
> Also did you test this gunicorn upgrade E2E with a live setup?

I did an upgrade by replacing the parcel directory on a ubuntu2204/py10 cluster. I still want to do a perf-test.

**Review by Harshg999 — APPROVED** [review]
LGTM!

**Review by ranade1 — APPROVED** [review]
have you compared the performance numbers?

**Pull request merged**